zkevm-circuits/src/evm_circuit.rs

```diff
@@ -408,32 +409,56 @@ pub mod test {
     }
 
     pub fn get_test_degree<F: Field>(block: &Block<F>) -> u32 {
+        let num_rows_required_for_execution_steps: usize =
+            EvmCircuit::<F>::get_num_rows_required(block);
+        let num_rows_required_for_rw_table: usize = block.circuits_params.max_rws;
+        let num_rows_required_for_fixed_table: usize = detect_fixed_table_tags(block)
+            .iter()
+            .map(|tag| tag.build::<F>().count())
+            .sum();
+        let num_rows_required_for_bytecode_table: usize = block
+            .bytecodes
+            .values()
+            .map(|bytecode| bytecode.bytes.len())
+            .sum();
+        let num_rows_required_for_copy_table: usize =
+            block.copy_events.iter().map(|c| c.bytes.len() * 2).sum();
+        let num_rows_required_for_keccak_table: usize = block.keccak_inputs.len();
```

Comment:
I think this one is not so straight forward.  `block.keccak_inputs` contains the list of keccak byte array inputs.  Each array is processed in chunks of 136 bytes (`const RATE` in keccak circuit).  For each chunk there are `NUM_ROUNDS + 1`, and each round takes `KECCAK_ROWS` (which can be accessed by `get_num_rows_per_round` in the keccak circuit module; it's 5 by default but can be changed via env var)

Reply:
that is true for the full keccak circuit. But for the table part, currently each byte input has only one row (RLC_input, RLC_hashoutput).    So this function is used to estimated rows for evm circuit, i think block.keccak_inputs.len() will be good.    When later we estimate keccak circuit rows in super circuit, then we should use `sum(KECCAK_ROWS * 25 * (Math.ceiling(bytes.len() / 136)) for bytes in block. keccak_inputs ]`

Reply:
Oh you're completely right!  My analysis doesn't apply in this EVM test context.